src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx

```diff
@@ -0,0 +1,576 @@
+        if (isExternal) {
+            try {
+                const parsedJson = JSON.parse(value)
+                return YAML.stringify(parsedJson, { indent: 2 })
+            } catch (e) {
+                return value
+            }
+        }
+        return value
+    }
+
+    function renderCodeEditor(): JSX.Element {
+        const version = deploymentHistoryArr[selectedDeploymentHistoryIndex].version
+        const selectedDeploymentManifestDetail = deploymentManifestDetails.get(version)
+
+        if (selectedDeploymentManifestDetail.loading && !selectedDeploymentManifestDetail.error) {
+            return <Progressing theme="white" pageLoader />
+        } else if (
+            !selectedDeploymentManifestDetail.loading &&
+            ((selectedDeploymentManifestDetail.error && selectedDeploymentManifestDetail.errorCode === 404) ||
+                (selectedDeploymentTabIndex === 1 && !selectedDeploymentManifestDetail.valuesYaml) ||
+                (selectedDeploymentTabIndex === 2 && !selectedDeploymentManifestDetail.manifest))
+        ) {
+            return (
+                <CDEmptyState
+                    subtitle={`${deploymentTabs[selectedDeploymentTabIndex]} is not available for this deployment`}
+                />
+            )
+        } else if (!selectedDeploymentManifestDetail.loading && selectedDeploymentManifestDetail.error) {
+            return (
+                <MessageUI
+                    iconClassName="error-exclamation-icon"
+                    theme="white"
+                    msg="There was an error loading the file."
+                    msgStyle={{ color: '#767D84', marginTop: '0' }}
+                    size={24}
+                    isShowActionButton={true}
+                    actionButtonText="Retry"
+                    onActionButtonClick={() => {
+                        checkAndFetchDeploymentDetail(version, true)
+                    }}
+                    actionButtonStyle={{ color: '#0066cc', textDecoration: 'none' }}
+                />
+            )
+        }
+        return (
+            <div className="bcn-0 border-btm">
+                <CodeEditor
+                    value={
+                        selectedDeploymentTabIndex === 1
+                            ? getEditorValue(selectedDeploymentManifestDetail.valuesYaml)
+                            : getEditorValue(selectedDeploymentManifestDetail.manifest)
+                    }
+                    noParsing
+                    mode="yaml"
+                    height={700}
+                    readOnly={true}
+                ></CodeEditor>
+            </div>
+        )
+    }
+
+    function renderSelectedDeploymentTabData() {
+        const deployment = deploymentHistoryArr[selectedDeploymentHistoryIndex]
+        const chartMetadata = deployment.chartMetadata
+
+        return (
+            <div className={`trigger-outputs-container ${selectedDeploymentTabIndex == 0 ? 'pt-20' : ''}`}>
```

Comment:
loader is missing here

Reply:
We won't be needing a loader here as we already have a loader at the parent level & data at this point is present.
On a side note - A loader has already been added while fetching the values.yaml when clicked on tab.